Cassie ODM CQL 3.0 Supports

vikramthyagarajan opened the pull request:
Using the new version of the cassandra driver (datastax/nodejs-driver) which has CQL 3.0 support and passed all npm tests. A few changes to tests are made, but no major rewrites.

Flux159 commented:
Going to test during the week and if everything's good, will merge and republish to npm as v0.2.0 with documentation updates.

vikramthyagarajan commented:
Hi @Flux159, any update on this?

Flux159 commented on lib/connection.js:
```diff
@@ -0,0 +1,61 @@
+      return callback();
+    }
+    db.batch(queryArray, options, callback);
+    // var batchedQuery = db.beginBatch();
```
Would be good to remove these comments (they're in git history anyways)

Flux159 replied:
This is currently breaking readme-tests/batch_test.js I think (run manually using "node readme-tests/batch_test.js")

I'm using node version v6.2.0 with Cassandra 3.9

Flux159 commented on lib/query.js:
```diff
@@ -96,13 +96,14 @@ function Query(queryString, params, connection, options, modelName) {
-            }
-        }
-    });
+    // hinting is not required, as the queries are prepared, which is the recommended way of fixing the issue
```
This line of comment is good, but the ones below could be removed

vikramthyagarajan commented:
I've fixed the `Error: Expected 4 or 0 byte int (8)` errors in readme-tests/batch_test.js and readme-tests/stream_test.js by preparing the batched and stream queries. Also, I've cleaned up the commented code that you'd mentioned.

Regarding the following error,
`ResponseError: Cannot page queries with both ORDER BY and a IN restriction on the partition key; you must either remove the ORDER BY or the IN and sort client side, or disable paging for this query`
The issue seems to be that the query is inefficient. I think cassandra recommends to solve it be using the clustering key option to be ASC, so that it can do away with the ORDER BY clause. The IN query itself would return it ordered by the clustering key. The query can still be run by disabling paging in CQLSH, but I haven't yet found a way to disable paging through the driver.

However, this would mean making these changes on the limit_sort_test-
1. Add clustering_key as name ASC
2. Remove sorting in the 2 find methods
Should I make and push these changes?

Flux159 commented:
Changes look good to me - you don't have to make the changes to limit sort, I'm going to do those after merging your changes in. Once that's done I'll publish 0.2.0 to npm as well.

Thanks a bunch for making these fixes.
